src/components/Inputs/DateRangePicker/DateRangePicker.vue

```diff
@@ -0,0 +1,913 @@
+        this.state.calendaryDates.startDate = date;
+      } else {
+        this.state.calendaryDates[dateType] = date;
+      }
+
+      this.state.isYearSelectorOpen[dateType] = false;
+    },
+
+    days(dateType, monthOffset = 0) {
+      const date = new Date(this.state.calendaryDates[dateType]);
+      date.setMonth(date.getMonth() + monthOffset);
+
+      const month = date.getMonth();
+      const year = date.getFullYear();
+
+      const firstDay = new Date(year, month, 1).getDay();
+      const daysInMonth = new Date(year, month + 1, 0).getDate();
+      const days = [...Array(daysInMonth).keys()].map(i => i + 1);
+
+      const dates = days.map((day, index) => {
+        const id = index + 1;
+        const dateObj = new Date(year, month, day);
+        return { id, date: dateObj };
+      });
+
+      const emptyDays = [...Array(firstDay).fill({ id: null, date: null })];
+      return [...emptyDays, ...dates];
+    },
+
+    selectDay(dateType, day) {
+      const { startDate, endDate } = this.state.inputValue;
+
+      if (day === null) return;
+
+      if (dateType === 'startDate') {
+        const newStartDate = new Date(this.state.calendaryDates.startDate);
+        newStartDate.setDate(day);
+
+        if (newStartDate > endDate) {
+          this.state.inputValue = {
+            startDate: newStartDate,
+            endDate: newStartDate,
+          };
+        } else {
+          this.state.inputValue = {
+            startDate: newStartDate,
+            endDate,
+          };
+        }
+      }
+
+      if (dateType === 'endDate') {
+        const newEndDate = new Date(this.state.calendaryDates.endDate);
+        newEndDate.setDate(day);
+
+        if (newEndDate < startDate) {
+          this.state.inputValue = {
+            startDate: newEndDate,
+            endDate: newEndDate,
+          };
+        } else {
+          this.state.inputValue = {
+            startDate,
+            endDate: newEndDate,
+          };
+        }
+      }
+    },
+
+    daysOnRange(day) {
+      const { startDate } = this.state.inputValue;
+      const { endDate } = this.state.inputValue;
+
+      const yesterday = startDate
+        ? new Date(startDate.getTime() - 86400000)
+        : new Date();
+
+      if (day >= yesterday && day <= endDate) return 'on-range';
+      if (day < startDate || day > endDate) return 'disabled';
+    },
+
+    changePeriod(period) {
+      const periodsInTimestamp = {
+        Hoje: {
+          startDate: new Date(),
+          endDate: new Date(),
+        },
+        'Últimos 30 dias': {
+          startDate: new Date(new Date().getTime() - 2592000000),
+          endDate: new Date(),
+        },
+        'Últimos 6 meses': {
+          startDate: new Date(new Date().getTime() - 15552000000),
+          endDate: new Date(),
+        },
+        'Últimos 12 meses': {
+          startDate: new Date(new Date().getTime() - 31104000000),
+          endDate: new Date(),
+        },
```

Comment:
Wouldn't be cool to use `decreaseDate` from adapcon-utils-js?
Is a simple function that receives a date and a certain duration do be decreased. I think its better than having numbers like this on the code

Reply:
Of course, I didn´t know about it, but I will change it.